### DISCUSSION TITLE
Add support for font antialiasing

### DISCUSSION THREAD

**Discussion by dgolub** [SixLabors/Fonts#256] [Ideas]
One major feature that's present in System.Drawing but missing in ImageSharp is rendering text with antialiasing.  I have some code on my web site for rendering county route shields that looks like this:

```
FontFamily family = SystemFonts.Get("Arial");
Font smallFont = family.CreateFont(3);
TextOptions smallTextOptions = new(smallFont) {
    Dpi = 96,
    HorizontalAlignment = HorizontalAlignment.Center,
    Origin = new Point(x, y),
    WrappingLength = 42
};
Color smallTextColor = Color.FromRgb(0xF7, 0xD1, 0x17);
image.Mutate(ctx => ctx.DrawText(smallTextOptions, "COUNTY", smallTextColor));
```

Due to the small 3-point font size, this leads to the text appearing pixelated as in the attached image.  I did not have this issue with System.Drawing when I enabled ClearType, but as far as I can tell, there's no equivalent in ImageSharp as the moment.

![cr97](https://user-images.githubusercontent.com/5299/159192949-44918a25-a4cd-476d-b51b-5d22d3a4deaa.png)

**Comment by JimBobSquarePants** [comment]
I think you're mixing AntiAliasing with Hinting.

Antialias is turned on by default.
https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.GraphicsOptions.html#SixLabors_ImageSharp_GraphicsOptions_Antialias

Hinting is turned off by default (because it's experimental) 
https://docs.sixlabors.com/api/Fonts/SixLabors.Fonts.TextOptions.html#SixLabors_Fonts_TextOptions_ApplyHinting

**Reply by dgolub** [reply]
No, that doesn't seem to make a difference.  I turned on `ApplyHinting`, and it still appears pixelated.

**Reply by JimBobSquarePants** [reply]
Like I said. Experimental....

Hinting is..... Hard. I'd love to learn a lot more about how FreeType does hinting but I find their code difficult to parse so would love some community help to improve matters.

**Reply by JimBobSquarePants** [reply]
Managed to make a fairly large improvement to the hinter (_not pushed currently_). I'll need to look at the drawing library next to figure out LCD ratserization to include the color anti aliasing System.Drawing uses. That's where a big improvement would come.

Hinting - Y direction only
![COUNTY](https://user-images.githubusercontent.com/385879/160239175-c8efb876-3aed-40ac-97a9-c353f09a5141.png)

Hinting - X & Y
![COUNTY](https://user-images.githubusercontent.com/385879/160239023-8e87d226-a6ba-493b-be84-2cae078b4aae.png)

**Reply by dgolub** [reply]
Awesome!  That definitely looks like a major improvement.  Looking forward to the upgrade!
